data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/ScanObjectWorker.java

```diff
@@ -0,0 +1,175 @@
+        this.s3Client = s3Client;
+        this.scanOptionsBuilderList = scanOptionsBuilderList;
+        this.s3ObjectHandler= s3ObjectHandler;
+    }
+
+    /**
+     * It will decide the s3 object parse <code>S3ObjectWorker</code> or <code>S3SelectWorker</code>
+     * based on s3 select configuration provided.
+     */
+    @Override
+    public void run() {
+        scanOptionsBuilderList.forEach(this::parseS3ScanObjects);
+    }
+
+    /**
+     * Method will parse the s3 object and write to {@link Buffer}
+     */
+    void parseS3ScanObjects(final ScanOptions scanOptions) {
+        scanIncludeObjectsValidationCheck(scanOptions);
+        scanOptions.getIncludeKeyPaths().forEach(key ->{
+            ListObjectsV2Request request = ListObjectsV2Request.builder()
+                    .bucket(scanOptions.getBucket())
+                    .prefix(key).build();
+
+            final ListObjectsV2Response s3ObjectResponse = s3Client.listObjectsV2(request);
+            s3ObjectResponse.contents().stream().map(S3Object::key).filter(s3ObjKey-> s3ObjKey.lastIndexOf(".")!=-1)
+                    .filter(s -> !scanOptions.getExcludeKeyPaths().contains(s.substring(s.lastIndexOf("."))))
+                    .forEach(s3Object ->
+                        processS3ObjectKeys(S3ObjectReference.bucketAndKey(scanOptions.getBucket(),
+                                s3Object).build(),s3ObjectHandler, scanOptions));
+        });
+    }
+
+    private static void scanIncludeObjectsValidationCheck(final ScanOptions scanOptions) {
+        if(Objects.isNull(scanOptions.getIncludeKeyPaths()) || Objects.isNull(scanOptions.getExcludeKeyPaths())){
+            throw new IllegalArgumentException("include/exclude list should not be empty in pipeline yaml configuration");
+        }
+    }
+
+    private void processS3ObjectKeys(final S3ObjectReference s3ObjectReference,
+                                     final S3ObjectHandler s3ObjectHandler,
+                                     final ScanOptions scanOptions){
+        final S3ObjectDetails s3ObjDetails = getS3ObjectDetails(s3ObjectReference);
+        final boolean isKeyMatchedBetweenTimeRange = isKeyMatchedBetweenTimeRange(s3ObjDetails.getS3ObjectLastModifiedTimestamp(),
+                scanOptions.getRange(),
+                scanOptions.getStartDateTime(),
+                scanOptions.getEndDateTime());
+        if(isKeyMatchedBetweenTimeRange && (isKeyProcessedByS3Scan(s3ObjDetails))){
+            updateKeyProcessedByS3Scan(s3ObjDetails);
+            try{
+                s3ObjectHandler.parseS3Object(s3ObjectReference,null);
+            }catch (IOException ex){
+                deleteKeyProcessedByS3Scan(s3ObjDetails);
+                LOG.error("Error while process the parseS3Object. ",ex);
+            }
+        }
+    }
+    /**
+     * Method will identify already processed key.
+     * @return boolean
+     */
+    public boolean isKeyProcessedByS3Scan(final S3ObjectDetails s3ObjectDetails) {
```

Comment:
This method returns true when key in not present in map (object is not processed). Should we rename this method `isKeyNotProcessedByS3Scan`?